src/Core/Managed/Shared/Extensibility/TelemetryConfiguration.cs

```diff
@@ -173,7 +245,26 @@ internal set
         /// <summary>
         /// Gets or sets the telemetry channel.
         /// </summary>
-        public ITelemetryChannel TelemetryChannel { get; set; }
+        public ITelemetryChannel TelemetryChannel
+        {
+            get
+            {
+                return this.telemetryChannel;
+            }
+
+            set
+            {
+                ITelemetryChannel oldChannel = this.telemetryChannel;
+                this.telemetryChannel = value;
+
+                // If we have a previously assigned channel which is not the same one as the "new" value
+                // passed in then we need to dispose of the old channel to keep from leaking resources.
+                if (oldChannel != null && oldChannel != value)
+                {
+                    oldChannel.Dispose();
```

Comment:
For DI it will give us the same instance over and over if it was registered as a singleton.
If it was not registered as a singleton then it will give a new instance each time.